libafl/src/events/llmp/hooks/centralized.rs

```diff
@@ -0,0 +1,117 @@
+use std::{fmt::Debug, marker::PhantomData};
+
+#[cfg(feature = "llmp_compression")]
+use libafl_bolts::{compress::GzipCompressor, llmp::LLMP_FLAG_COMPRESSED};
+use libafl_bolts::{
+    llmp::{Flags, LlmpHook, LlmpMsgHookResult, Tag},
+    shmem::ShMemProvider,
+    ClientId, Error,
+};
+
+#[cfg(feature = "llmp_compression")]
+use crate::events::COMPRESS_THRESHOLD;
+use crate::{
+    events::{BrokerEventResult, Event, _LLMP_TAG_TO_MAIN},
+    inputs::Input,
+};
+
+/// An LLMP-backed event manager for scalable multi-processed fuzzing
+pub struct CentralizedLlmpHook<I, SP>
+where
```

Comment:
You never need to add type constraint to `pub struct`

Reply:
Ok, but not sure removing bounds completely from structs helps for readability.
It's not necessary, but it's very unlikely you will ever want to use the struct with something else than the main bound(s) you have in mind.
I think it's good to actually see what the kind of object we want to work with is like directly, instead of having to scroll until the actual place we put the bound, but maybe there's a reason i didn't notice?

Reply:
hum I see, it's a good point. if we make sure to keep the bounds consistent ok.
i'll propagate the changes.